### DISCUSSION TITLE
In-app API Explorer for a better DX

### DISCUSSION THREAD

**Discussion by AnishDe12020** [directus/directus#13054] [Feature Requests]
It would be awesome if Directus had something like an in-app API explorer. This could be something similar to a GraphQL schema explorer but it can also be extended into providing queries for REST, GraphQL, and even code snippets for the JS SDK. It can also include the ability to send API requests and see the response. In all, it should provide a better DX

Would love to hear what the team has to say about this

**Comment by benhaynes** [comment]
I agree, it would be great to have a full-featured API explorer built in to the App. We'd need to find a good white-label, unopinionated library with the right license and features, or look into building one (a _much_ bigger effort).

**Reply by rijkvanzanten** [reply]
I don't think any library exists that works like that for both rest and graphql. GraphQL has GraphiQL (and others), REST is a little more opinionated, but could be made to work through the Swagger/OpenAPI stuff, though they don't necessarily support all the things that our REST api can do 🤔 I think it'll have to be a custom thing

**Reply by benhaynes** [reply]
Down for looking into a custom MVP solution for this.

**Reply by AnishDe12020** [reply]
So, i just quicly thought of an approach to this.

The component could have a sidebar with the schema for our current app (like graphiql). Whenever one toggles the checkboxes, a GraphQL query gets formed on the right side. (also just like graphiql). Then one can probably choose REST or SDK (probably can be a tabbed layout) and it will just convert the query to the respective language.

**Reply by sxsws** [reply]
hasura.io and strapi have api explorers so maybe the sources can be ripped out?

**Reply by rijkvanzanten** [reply]
> hasura.io and strapi have api explorers so maybe the sources can be ripped out?

Great for inspiration, but I would never rip/copy-paste anything they've done 👍🏻

**Reply by sxsws** [reply]
no, I mean, I'm assuming they're using existing libraries for their API explorer

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
